SD_Assignment1/src/main/java/ro/utcn/aron/persistence/jdbc/JdbcQuestionRepository.java

```diff
@@ -0,0 +1,409 @@
+	}
+
+	private List<String> getTagsFromJdbc(int questionId) {
+		return template.query("SELECT * FROM tags WHERE questionid = ?",
+				(resultSet, i) -> new String(resultSet.getString("tags")), questionId);
+	}
+
+	private String list2String(List<String> tagList) {
+		String text = "";
+		for (int i = 0; i < tagList.size() - 1; i++) {
+			text += tagList.get(i) + ",";
+		}
+		text += tagList.get(tagList.size() - 1);
+
+		return text;
+	}
+
+	@Override
+	public void answerQuestion(int questionid, String user, String answer) {
+		SimpleJdbcInsert insert = new SimpleJdbcInsert(template);
+		insert.setTableName("answers");
+		insert.setGeneratedKeyName("answerid");
+
+		Map<String, Object> data = new HashMap<>();
+		data.put("questionid", questionid);
+		data.put("author", user);
+		data.put("answer", answer);
+
+		data.put("creationdate", new Date().toString());
+		insert.execute(data);
+	}
+
+	private List<Answer> getAnswersFromJdbc(int questionid) {
+		return template.query("SELECT * FROM answers WHERE questionid = ?",
+				(resultSet, i) -> new Answer(resultSet.getInt("answerid"),resultSet.getString("answer"), resultSet.getString("author"),
+						resultSet.getString("creationdate"), getAnswerScore(resultSet.getInt("answerid")) ),
+				questionid);
+	}
+	
+	private int getAnswerScore(int answerid) {
+		return template.query("SELECT * FROM avotes WHERE answerid = ?",
+				(resultSet, i) -> new Vote(resultSet.getString("author"), resultSet.getInt("voteid"), resultSet.getInt("votetype")),
+				answerid).stream().mapToInt(vote->vote.getVoteYype()).sum();
+	}
+	
+	private int getQuestionScore(int questionid) {
+		return template.query("SELECT * FROM qvotes WHERE questionid = ?",
+				(resultSet, i) -> new Vote(resultSet.getString("author"), resultSet.getInt("voteid"), resultSet.getInt("votetype")),
+				questionid).stream().mapToInt(vote->vote.getVoteYype()).sum();
+	}
+
+	@Override
+	public void editQuestion(int questionid, String user, String text) {
+		List<Question> result = template.query("SELECT * FROM question WHERE id = ? and author = ?", (resultSet,
+				i) -> new Question(resultSet.getInt("id"), resultSet.getString("title"), resultSet.getString("body"),
+						list2String(getTagsFromJdbc(resultSet.getInt("id"))), resultSet.getString("author"),
+						resultSet.getString("creationdate"),getAnswersFromJdbc(resultSet.getInt("id"))), questionid, user);
+		
+		if(result.isEmpty()) {
+			System.out.println("You cannot edit other question!");
+			return;
+		}
+		
+		template.update("UPDATE question SET body = ? WHERE id = ? ",
+				text, questionid);
+	}
+
+	@Override
+	public void removeAnswer(int answerid, String user) {
+		List<Answer> result = template.query("SELECT * FROM answers WHERE answerid = ? and author = ?", (resultSet,
+				i) -> new Answer(resultSet.getInt("answerid"), resultSet.getString("author"),
+						 resultSet.getString("answer"),
+						resultSet.getString("creationdate")), answerid, user);
+		
+		if(result.isEmpty()) {
```

Comment:
Same as above. Basically you should have in your repositories methods for executing these queries and returning the results; but the logic checks should be in the service layer.